pages/index.js

```diff
@@ -18,7 +18,10 @@ import useStore from "@/store";
 
 export default function HomePage({ onSubmit }) {
   const { data: session, status } = useSession();
-  const { data: activities, isLoading } = useSWR("/api/activities");
+  const { data: activities, isLoading: activitiesIsLoading } =
```

Comment:
I think the fact that you have renamed the variable from `isLoading` to `activitiesIsLoading` will have knock on effects further down in your code. 

i.e. The code below references `isLoading` but that variable no longer exists. 

``` js
if (isLoading) return <div>loading...</div>;
```

In general it feels like you are handling the loading states multiple times. It is probably time to review this and tidy things up a bit.